### PR TITLE
docs: README updates

### LINKED ISSUE #726: miscellaneous documentation issues
## Bug Description

There are a few areas of the docs that are wrong or need clarification.

1. >where each line contains a JSON encoded array

Now there are metadata lines as well

https://github.com/IBM/couchbackup/blob/7cd6a7551c8cf041acec621f227d4155c84f9dfa/README.md?plain=1#L177

2. >up to `buffer-size` objects

This can be more than `buffer-size` objects because of the way revs are returned

https://github.com/IBM/couchbackup/blob/7cd6a7551c8cf041acec621f227d4155c84f9dfa/README.md?plain=1#L177

3. >No revision tokens are saved 

Since version [`2.4.2`](https://github.com/IBM/couchbackup/releases/tag/2.4.2) revisions have been saved for shallow backups.

https://github.com/IBM/couchbackup/blob/7cd6a7551c8cf041acec621f227d4155c84f9dfa/README.md?plain=1#L195
https://github.com/IBM/couchbackup/blob/7cd6a7551c8cf041acec621f227d4155c84f9dfa/README.md?plain=1#L220

4. >Shallow backups cannot be resumed because they do not produce a log file.

I had a report of some customer confusion around this statement. Let's just say the `--log` and `--resume` options are invalid for shallow backups.

https://github.com/IBM/couchbackup/blob/7cd6a7551c8cf041acec621f227d4155c84f9dfa/README.md?plain=1#L196

5. >These can only be ignored

This statement only applies to backups created with versions older than 2.10.0, since then errors cannot be ignored.

https://github.com/IBM/couchbackup/blob/main/README.md?plain=1#L364-L366

6. >failed requests are retried indefinitely

Not so, there is a limit on the retries.

https://github.com/IBM/couchbackup/blob/7cd6a7551c8cf041acec621f227d4155c84f9dfa/README.md?plain=1#L402

7. >For non-fatal errors an `error` event will be emitted

There are no non-fatal errors any more.

https://github.com/IBM/couchbackup/blob/7cd6a7551c8cf041acec621f227d4155c84f9dfa/README.md?plain=1#L410

8. Missing error code LogFileExists: 23,

https://github.com/IBM/couchbackup/blob/7cd6a7551c8cf041acec621f227d4155c84f9dfa/README.md?plain=1#L430

### PR DESCRIPTION
**Pull request by ricellis**
## Checklist

- [x] Added tests for code changes _or_ test/build only changes
- [x] Updated the change log file (`CHANGES.md`) _or_ test/build only changes
- [x] Completed the PR template below:

## Description

Fixes #726

## Approach

* Fix errors from #726.
* Fix some typos.
* Clean up a lot of Vale warnings.

## Schema & API Changes

- "No change"

## Security and Privacy

- "No change"

## Testing

- N/A docs

## Monitoring and Logging

- "No change"

### CODE REVIEW

**Review by eiri — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -169,64 +169,70 @@ couchbackup --db animaldb | openssl aes-128-cbc -pass pass:12345 > encrypted_ani
...
 
-## What is shallow mode?
+## What's shallow mode?
 
-When you run `couchbackup` with `--mode shallow` a simpler backup is performed, only backing up the winning revisions
-of the database. No revision tokens are saved and any conflicting revisions are ignored. This is a faster, but less
-complete backup. Shallow backups cannot be resumed because they do not produce a log file.
+When you run `couchbackup` with `--mode shallow` couchbackup performs a simpler backup.
+It only backs up the winning revisions and ignores any conflicting revisions.
+This is a faster, but less complete backup. The `--log` and `---resume` options
```
Comment (emlaver) [inline README.md:202]: I wonder if we should move below to a note:
```
**Note:** The `--log` and `---resume` options are not valid for shallow mode backups.
```

File: README.md [hunk 2]
```diff
@@ -169,64 +169,70 @@ couchbackup --db animaldb | openssl aes-128-cbc -pass pass:12345 > encrypted_ani
...
+ are not valid for shallow mode backups.
 
-NOTE: Parallellism will not be in effect if `--mode shallow` is defined.
+NOTE: Parallelism does not apply for `--mode shallow`.
```
Comment (emlaver) [inline README.md:205]:
```suggestion
**Note:** Parallelism does not apply for `--mode shallow`.
```

File: README.md [hunk 3]
```diff
@@ -437,13 +440,13 @@ details them.
 
 ## Note on attachments
 
-TLDR; If you backup a database that contains attachments you will not be able to restore it.
+TLDR; If you backup a database that has attachments couchrestore cannot restore it.
```
Comment (emlaver) [inline README.md:443]:
```suggestion
TLDR; If you backup a database that has attachments `couchrestore` cannot restore it.
```

**Review by emlaver — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (emlaver) [reply README.md:205]: If you think the comment above is worthwhile, we could probably put this all on one line:
```suggestion
**Note:** The `--log`,  `---resume`, and `--parallelism` options are not valid for shallow mode backups.
```

File: README.md (hunk 3 above)
Comment (ricellis) [reply README.md:443]: We have a mix of plain text and monospaced throughout the README, would you like to see `couchbackup` and `couchrestore` everywhere?

File: README.md (hunk 1 above)
Comment (ricellis) [reply README.md]: Done in 8252f5d

File: README.md (hunk 3 above)
Comment (emlaver) [reply README.md:442]: Yea, would be nice to have monospaced everywhere but up to you

Comment (ricellis) [reply README.md]: Done in 56a61b2 (I also removed  a TXE caveat that is obviously no longer relevant).

Comment (emlaver) [reply README.md]: Thanks and good catch!
